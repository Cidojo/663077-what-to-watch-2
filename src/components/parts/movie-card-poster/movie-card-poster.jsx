import * as React from 'react';
import * as PropTypes from 'prop-types';

const MovieCardPoster = (props) => {
  const {extraClassName, name, posterImage} = props;

  return (
    <div className={`movie-card__poster ${extraClassName}`}>
      <img
        src={posterImage}
        alt={name}
        width="218"
        height="327"
      />
    </div>
  );
};

MovieCardPoster.propTypes = {
  name: PropTypes.string.isRequired,
  posterImage: PropTypes.string.isRequired,
  extraClassName: PropTypes.string
};

MovieCardPoster.defaultProps = {
  name: ``,
  posterImage: ``,
  extraClassName: ``
};

export {MovieCardPoster};
