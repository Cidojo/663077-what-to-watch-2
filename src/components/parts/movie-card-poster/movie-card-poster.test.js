import * as React from 'react';
import {MovieCardPoster} from './movie-card-poster.jsx';
import * as renderer from 'react-test-renderer';

const props = {
  name: ``,
  posterImage: ``,
  extraClassName: ``
};

it(`should render MovieCardPoster component without errors`, () => {
  const tree = renderer.create(<MovieCardPoster {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
