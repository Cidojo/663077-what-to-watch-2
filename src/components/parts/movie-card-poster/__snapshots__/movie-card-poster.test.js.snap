// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MovieCardPoster component without errors 1`] = `
<div
  className="movie-card__poster "
>
  <img
    alt=""
    height="327"
    src=""
    width="218"
  />
</div>
`;
